Discord-Tip-Bot/src/database/control.ts

```diff
@@ -0,0 +1,88 @@
+import JoyModel from "./models";
+
+export const getJoyData = async (username: string) => {
+    const JoyData =
+        (await JoyModel.findOne({ userName: username })) ||
+        (await JoyModel.create({
+            userName: username,
+            walletAddress: 0,
+            amount: 0,
+            day: Date.now(),
+            collageAmount: 0,
+        }));
+
+    return JoyData;
+};
+
+export const updateJoyData = async (userName: string, amount: number) => {
+    const joyData = await JoyModel.findOne({ userName: userName });
+    if (!joyData) return "You are not registered. Please register now.";
+
+    joyData.amount = amount;
+    joyData.day = Date.now();
+    joyData.collageAmount ? joyData.collageAmount += amount : joyData.collageAmount = amount;
+
+    joyData.save();
+    return `Your deposit is ${amount} JOY`;
+
+};
+
+export const setJoyData = async (userName: string, address: string) => {
+    const JoyData =
+        (await JoyModel.findOne({ userName: userName })) ||
+        (await JoyModel.create({
+            userName: userName,
+            walletAddress: address,
+            amount: 0,
+            day: Date.now(),
+            collageAmount: 0,
+        }));
+    JoyData.walletAddress = address;
+    return JoyData.save();
+};
+
+export const sendJoyToken = async (userName: string, reiceve: string, amount: number) => {
+    let error: string;
+
+    const tx = await JoyModel.findOne({ userName: userName });
+    if (!tx) return error = "Error : sender's username is unregistered";
+
+    const rx = await JoyModel.findOne({ userName: reiceve });
+    if (!rx) return error = "Error : receiver's username is unregistered";
+
+
+    const sendJoy = await JoyModel.findOne({ userName: userName });
+    if (!sendJoy) return
+
+    sendJoy.amount = amount;
+    sendJoy.day = Date.now();
+    sendJoy.collageAmount -= amount;
+
+    sendJoy.save();
+
+    const recieveJoy = await JoyModel.findOne({ userName: reiceve });
```

Comment:
Why do we fetch the same users twice in this function?

Reply:
It is necessary to verify that the sender address and receiver address are valid. Without this, the sender will lose JOY if the receiver address is invalid

Reply:
That is necessary. The code is verifying whether the sender address and receiver address are valid. If this verification process is skipped and the receiver address is invalid, the sender would lose JOY.

Reply:
Sorry if I'm missing something obvious but didn't we already perform this same check at the top of the function (for both the sender and receiver)? What is stopping us from just using `tx` and `rx` instead of `sendJoy` and `receiveJoy` 🤔

Reply:
You are right. Just fixed the code. :)